### ISSUE TITLE
Commands and Percentages off or is it my commands?

### ISSUE THREAD

**Issue by DavidV-WPCK** [guideguide/support#139]
I have several commands to recreate a mobile first responsive grid.
Example:
| 32.222222222222% | 2.5% | 66.111111111111% | ( vl, ~ | 1200px | ~ )

Unless I'm wrong this should create the 1/3rd | gutter | 2/3rds 1200px wide centered on a 1400px wide canvas.
The result I get is alway off by either a few pixels or more. In this example 10px.
![screenshot jun 16 15 38 45](https://cloud.githubusercontent.com/assets/4623750/8195167/d368e8de-143d-11e5-891d-c88264250fdd.png)

You can see the guide at 1310px (should be at 1300px).

Basically I use 100px on the left and right of the canvas so I have some room to breath.

If I do a 1/2 | 1/2 grid it does work out just right (guides at 100px and 1300px)
Example: 
| 48.75% | 2.5% | 48.75% | ( vl, ~ | 1200px | ~ )

Am I not using the commands correctly perhaps?
I'm trying to stick with the exact percentages in my grid, but maybe I should convert them to pixels?

Love this extension!

**Comment by cameronmcefee** [comment]
Hi @DavidV-WPCK,

I :heart: seeing people use grid notation, even if it doesn't work as expected. I'll walk you through what's going on. First of all, if you're curious about the really heinous math/code that it takes to calculate it, [check it out here](https://github.com/guideguide/grid-notation/blob/master/src/grid-notation.coffee).

The short answer is that your percentages add up to `100.833333333333%`, which equates to 1210px. In your second example, they add up to `100%`, which is why it works as expected.

To give you a rough idea of the math going on, GuideGuide first measures the whole distance available, then subtracts any exact measurement, then divides the remainder amongst the percentage. To give an example, if you were to specify margins and a 3 column grid using 33.33%, GuideGuide would subtract the width of the margins from the total width, then make each column 33.33% of the remainder. The subtraction portion isn't important in your case, but is worth mentioning because it modifies what a percentage means in most cases.

Now, as for the inconsistency.

- First, the sum of the percentages is 100.833333333333%. That's more than 100%. 

- Next, you're likely using pixel snapping. Percentages usually yield fractions of pixels, so GuideGuide has to round the numbers to ensure you have full integer pixel coordinates for your guides.

- You've specified that the grid should be center aligned.

To do the math:

- `Round down (.32222222222222*1200)` = 386
- `Round down (.025*1200)` = 30
- `Round down (.66111111111111*1200)` = 793
- `386 + 30 + 793` = 1209
- `(1400-1200)/2` = 100 offset to center (When you set an exact width, GuideGuide uses it for positioning).

So that ends up putting your first guide at 100, and the last one at 1310 (I think the rounding adds an extra pixel in there, hence the discrepancy from my math above.

**Comment by DavidV-WPCK** [comment]
Wow @cameronmcefee , thank you for the extensive reply!
Yeah, I love the grid notation too, it's a fantastic addition to guideguide, and the sets!
Hah... I am pixel snapping.
Really appreciate the breakdown, I'll get this worked out now. [guideguide](https://github.com/guideguide)  is such a huge time saver! Thanks.

**Comment by cameronmcefee** [comment]
:+1: I'm closing this one as "resolved" but feel free to reopen if you think there is a further issue to discuss.
